domain/secret/service/service.go

```diff
@@ -334,10 +334,10 @@ func (s *SecretService) CreateCharmSecret(ctx context.Context, uri *secrets.URI,
 // the secret owner already has a secret with the same label.
 // It returns [secreterrors.PermissionDenied] if the secret cannot be managed by the accessor.
 func (s *SecretService) UpdateUserSecret(ctx context.Context, uri *secrets.URI, params UpdateUserSecretParams) (errOut error) {
-	if err := s.canManage(ctx, uri, params.Accessor, nil); err != nil {
+	isLeader, err := checkLeaderToken(nil, params.Accessor)
```

Comment:
For user secrets we know leadership is not relevant. So I think we can just pass in false to the canManage call.